qubes/api/admin.py

```diff
@@ -336,6 +354,49 @@ def vm_volume_import(self):
 
         return '{} {}'.format(size, path)
 
+    @qubes.api.method('admin.vm.tag.List', no_payload=True)
+    @asyncio.coroutine
+    def vm_tag_list(self):
+        assert not self.arg
+
+        tags = self.dest.tags
+
+        tags = self.fire_event_for_filter(tags)
+
+        return ''.join('{}\n'.format(tag) for tag in sorted(tags))
+
+    @qubes.api.method('admin.vm.tag.Get', no_payload=True)
+    @asyncio.coroutine
+    def vm_tag_get(self):
+        qubes.vm.Tags.validate_tag(self.arg)
+
+        self.fire_event_for_permission()
+
+        return '1' if self.arg in self.dest.tags else '0'
```

Comment:
Having convention of `'0'` being false is a footgun for Python and requires some precautions  C. f. `qubes.property.bool`: we were lucky in that case because there is no easy way to get around that setter. In this case it is much easier. And IIRC we already have some places when `False`/`None` gets written as zero-length.

This particular call seems to be only used for `__contains__` dunder in core-admin-client, so this one seems innocent, however I'd be wary of introducing this in more places that needed.